core/crypto/src/traits.rs

```diff
@@ -62,17 +62,6 @@ macro_rules! common_conversions {
 
 macro_rules! common_conversions_fixed {
     ($ty:ty, $l:literal, $bytes:expr, $what:literal) => {
-        impl ::std::convert::TryFrom<&[u8]> for $ty {
-            type Error = ();
-
-            fn try_from(value: &[u8]) -> Result<Self, ()> {
-                match value.len() {
-                    $l => Self::try_from(::arrayref::array_ref!(value, 0, $l)).or(Err(())),
```

Comment:
I think we could in general get rid of `array_ref`, today this functionality is handled by the `TryFrom`s

Reply:
Yes, I think so. I have some wip commits doing that somewhere.

PS. Can I haz S-automerge.  I can no longer change labels.